pennylane/collections/apply.py

```diff
@@ -16,6 +16,7 @@
 """
 
 
+# pylint:disable=unnecessary-lambda-assignment
```

Comment:
It's weird that these are needed to make `pylint` pass locally but not in CI :thinking: Is this a version mismatch?

Reply:
tbh I just added them to not have to spend time figuring out what was causing the difference, but I'll check if a version difference is causing it

Reply:
it made some differences but not all.. I'm really not sure but I'm personally happy to add these ignores and call it a day (they are sensible after all)